src/engine/enginebuffer.cpp

```diff
@@ -1163,12 +1163,24 @@ void EngineBuffer::processSeek(bool paused) {
             return;
     }
 
+    double syncPosition = position;
     if (!paused && (seekType & SEEK_PHASE)) {
-        position = m_pBpmControl->getNearestPositionInPhase(position, true, true);
+        syncPosition = m_pBpmControl->getNearestPositionInPhase(position, true, true);
     }
-    if (position != m_filepos_play) {
-        setNewPlaypos(position, adjustingPhase);
+
+    if (syncPosition != m_filepos_play) {
+        if(seekType & SEEK_PHASE){
+            // inform controllers about the seek position itself
+            // and do the phase shift afterwards
+            // this prevents loops from getting disabled because the synced position is in front of the loop
+            setNewPlaypos(position, false);
+            adjustingPhase = true;
+        }
+        setNewPlaypos(syncPosition, adjustingPhase);
```

Comment:
The double setNewPlaypos() call reads a bit like a hack. Can we add for example a new parameter to identify the situation explicit?

Reply:
Yep, this is more like a workaround but as I am still new to the project I didn't want to change parameters of already existing methods. We could have a seekedPosition and syncedPosition in setNewPlaypos and notifySeek and let the controller decide what to do with this information. How about that?